x-pack/plugins/watcher/public/sections/watch_edit/components/logging_action_fields.tsx

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { Fragment } from 'react';
+import { EuiFieldText } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { ErrableFormRow } from '../../../components/form_errors';
+import { LoggingAction } from '../../../../common/types/action_types';
+
+interface Props {
+  action: LoggingAction;
+  editAction: (changedProperty: { key: string; value: string }) => void;
+  errors: { [key: string]: string[] };
+  hasErrors: boolean;
+}
+
+export const LoggingActionFields: React.FunctionComponent<Props> = ({
+  action,
+  editAction,
+  errors,
+  hasErrors,
+}) => {
+  const { text } = action;
+  return (
+    <Fragment>
+      <ErrableFormRow
+        id="loggingText"
+        errorKey="text"
+        fullWidth
+        errors={errors}
+        isShowingErrors={hasErrors && text !== undefined}
+        label={i18n.translate(
+          'xpack.watcher.sections.watchEdit.threshold.loggingAction.logTextFieldLabel',
+          {
+            defaultMessage: 'Log text',
+          }
+        )}
+      >
+        <EuiFieldText
+          fullWidth
+          name="text"
+          value={text || ''}
+          onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
+            editAction({ key: 'text', value: e.target.value });
+          }}
+          onBlur={() => {
```

Comment:
This was added as part of the validation implementation. If a field is required, it won’t be marked as so until after the user leaves an input field.

Reply:
side note - i noticed for the SR work, the fields are validated after the user submits, rather than immediately. is there a preferred approach?

Reply:
> side note - i noticed for the SR work, the fields are validated after the user submits, rather than immediately. is there a preferred approach?

Form validation UX is tough to get right and I think what we're doing elsewhere works pretty well but I'm not at all confident that it's ideal.I don't think it matters much that it works slightly differently than we have in other UIs. Given that the UX seems to work pretty well, I don't think we need to spend more time refining it in this iteration.

>  If a field is required, it won’t be marked as so until after the user leaves an input field.

Could you add a comment describing this intention here and in other places where we use an onBlur this way?